### DISCUSSION TITLE
MSI Silent Install with Changed Directory

### DISCUSSION THREAD

**Discussion by ghost** [StrawberryPerl/Perl-Dist-Strawberry#189] [Q&A] [answered]
Hi folks,

Let's just say its because I'm sick as to why I can't work this out today!!  :P

Can someone point out to me where I can or why I can't, get the directory changed for silent installation via MSI please ??

I can do it in the MSI's GUI, but I need the change via a silent install. And as nice as using Strawberry is, I'd rather use StrawberryPerl for our environment please. I've tried changing, via a transform file, the two (2) public property fields, but still my change does not work; WIXUI_INSTALLDIR and WixUI_Mode.

Any help greatly appreciated.

Kind regards,
Daniel.

**Comment by shawnlaffan** [comment, marked as answer]
Untested, but perhaps the responses at this post are useful?
https://stackoverflow.com/questions/8839978/how-do-you-install-an-msi-with-msiexec-into-a-specific-directory

An alternative is to use a portable strawberry distribution and update the paths as needed.  See the portableshell.bat file at the top of each such distribution.

**Reply by ghost** [reply]
Fantastic!
Thanks for that info @shawnlaffan  -  that did the trick.

Something tells me that I should have remembered or known that already
INSTALLDIR and TARGETDIR do seem familiar to me 🤔


Interestingly:

 > INSTALLDIR switch worked for me right off the bat and installed to my custom directory

 > TARGETDIR switch did not; it created my custom directory, but then right after doing so, proceeded to create the default directory and installed to it!

 > INSTALLLOCATION switch did not work

 > INSTALLFOLDER switch did not work

> APPLICATIONFOLDER switch did not work

 > DEFAULTPATHC switch did not work

 > APPDIR switch did not work

 > INSTALLPATH switch did not work
 
 and

 > ARPINSTALLLOCATION switch did not work
